tests/lib.rs

```diff
@@ -133,10 +134,15 @@ impl VirtualFileSystem {
 
             println!("\n-- {} –-\n", file.display());
 
-            assert_eq!(
-                &fs::read(&expected).expect("file not found"),
-                &fs::read(&actual).expect("file not found"),
-            );
+            match fs::read_to_string(&expected) {
+                Ok(expected_text) => assert_str_eq!(
+                    expected_text,
+                    fs::read_to_string(&actual).expect("file not found")
+                ),
+                // TODO(amello): make tgz deterministic so we can validate binary files
+                Err(err) if matches!(err.kind(), ErrorKind::InvalidData) => (),
```

Comment:
Can we filter tar / tgz files from the list of files to compare instead of catching this error?